github/github.go

```diff
@@ -880,6 +880,32 @@ func cloneRequest(r *http.Request) *http.Request {
 	return r2
 }
 
+// formatRateLimitResetDuration formats d to look like "[rate reset in 2s]" or
+// "[rate reset in 87m02s]" for the positive durations. And like "[rate limit was reset 87m02s ago]"
+// for the negative cases.
+func formatRateLimitResetDuration(d time.Duration) string {
```

Comment:
It's good that the func name is descriptive and specific to the task (since it's only being used in one place), but I think we can shorten it to just `formatRateReset`.

It happens to accept a `time.Duration` parameter, but that doesn't mean we need to put "Duration" in its name. Rate reset should be descriptive enough, since it's being passed `r.Rate.Reset` and it's what we call it in the string.